CorpusCallosum/fastsurfer_cc.py

```diff
@@ -0,0 +1,880 @@
+
+    return args
+
+
+def centroid_registration(aseg_nib: nib.Nifti1Image, verbose: bool = False) -> tuple[
+    np.ndarray, np.ndarray, np.ndarray, nib.Nifti1Header, np.ndarray
+]:
+    """Perform centroid-based registration between subject and fsaverage space.
+
+    Computes a rigid transformation between the subject's segmentation and fsaverage space
+    by aligning centroids of corresponding anatomical structures.
+
+    Parameters
+    ----------
+    aseg_nib : nibabel.Nifti1Image
+        Subject's segmentation image.
+    verbose : bool, optional
+        Whether to print progress information, by default False.
+
+    Returns
+    -------
+    orig_fsaverage_vox2vox : np.ndarray
+        Transformation matrix from original to fsaverage voxel space.
+    orig_fsaverage_ras2ras : np.ndarray
+        Transformation matrix from original to fsaverage RAS space.
+    fsaverage_hires_affine : np.ndarray
+        High-resolution fsaverage affine matrix.
+    fsaverage_header : nibabel.Nifti1Header
+        FSAverage header fields for LTA writing.
+    vox2ras_tkr : np.ndarray
+        Voxel to RAS tkr-space transformation matrix.
+
+    Notes
+    -----
+    The function uses pre-computed fsaverage centroids and data from static files
+    to perform the registration. It matches corresponding anatomical structures
+    between the subject's segmentation and fsaverage space.
+    """
+    if verbose:
+        print("Centroid registration")
+
+    # Load pre-computed fsaverage centroids and data from static files
+    centroids_dst = load_fsaverage_centroids(FSAVERAGE_CENTROIDS_PATH)
+    fsaverage_affine, fsaverage_header, vox2ras_tkr = load_fsaverage_data(FSAVERAGE_DATA_PATH)
+
+    centroids_mov, ids_not_found = get_centroids_from_nib(aseg_nib, label_ids=list(centroids_dst.keys()))
+
+    # delete not found labels from centroids_mov
+    for id in ids_not_found:
+        del centroids_dst[id]
+
+    centroids_mov = np.array(list(centroids_mov.values())).T
+    centroids_dst = np.array(list(centroids_dst.values())).T
+
+    orig_fsaverage_ras2ras = find_rigid(p_mov=centroids_mov.T, p_dst=centroids_dst.T)
+
+    # make affine that increases resolution to orig resolution
+    resolution_orig = aseg_nib.header.get_zooms()[0]
+    resolution_trans = np.eye(4)
+    resolution_trans[0, 0] = resolution_orig
+    resolution_trans[1, 1] = resolution_orig
+    resolution_trans[2, 2] = resolution_orig
+
+    orig_fsaverage_vox2vox = (
+        np.linalg.inv(resolution_trans @ fsaverage_affine) @ orig_fsaverage_ras2ras @ aseg_nib.affine
+    )
+    fsaverage_hires_affine = resolution_trans @ fsaverage_affine
+
+    return orig_fsaverage_vox2vox, orig_fsaverage_ras2ras, fsaverage_hires_affine, fsaverage_header, vox2ras_tkr
+
+
+def localize_ac_pc(
+    midslices: np.ndarray,
+    aseg_nib: "nib.Nifti1Image",
+    orig_fsaverage_vox2vox: np.ndarray,
+    model_localization: "torch.nn.Module",
```

Comment:
Since we use `run_model_on_slice`below, the type specification should probably be more specific. I Assume we could just add the respective Model from `localization_inference.py`?